### ISSUE TITLE
Removing 127.0.0.1 localhost from host file cause problems

### ISSUE THREAD

**Issue by TeaBough** [CiscoCloud/microservices-infrastructure#689]
I have a vagrant cluster with 3 leaders and when `consul-wait-for-leader.sh` is run it says that `No Consul leader elected in 30 seconds`, it is because it tries to curl `localhost:8500` the problem is that it can't resolv localhost since it has been remove by the common role. I dont' understand why we do this :

```
- name: clean hosts file
  sudo: yes
  lineinfile:
    dest: /etc/hosts
    regexp: "{{ item }}"
    state: absent
  with_items:
    - '^127\.0\.0\.1(\s+){{ inventory_hostname }}.*'
    - '^::1(\s+){{ inventory_hostname }}.*'
```

**Comment by eirslett** [comment]
In my project, I still get a 127.0.0.1 -> localhost reference in /etc/hosts. I suspect the regexp stuff is quite brittle...

**Comment by TeaBough** [comment]
When I `vagrant up --no-provision` my `/etc/hosts` looks like this :

```
127.0.0.1 worker1 localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdom
```

Thus the regex `^127\.0\.0\.1(\s+){{ inventory_hostname }}.*` match the first line. It's odd that in your case it doen't match.

**Comment by eirslett** [comment]
One option is to remove the regexp stuff altogether, and instead build the hosts file from scratch. That would be more deterministic...@ChrisAubuchon do you have any thoughts on that? What was the reason the regexp solution was chosen over a template solution?

**Comment by ChrisAubuchon** [comment]
My `vagrant up --no-provision1 has `/etc/hosts`:
```
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
```

The `worker1` in you `/etc/hosts` file is the reason the regex matches yours. Having the hostname as an alias for localhost breaks a few things.

I suspect the reason the regex was used was to try to just remove `{{ inventory_hostname }}` as an alias to localhost. Might be best to recreate from scratch using a template

**Comment by loshkovskyi** [comment]
> I dont' understand why we do this:

@TeaBough it's becase there were some duplicated `127.0.0.1` entries pointed to `{{ inventory_hostname }}` added in some OpenStack environments. More info in this commit: e9514ea050990258bb3375f0c517c405e0fc976c

**Comment by ryane** [comment]
fixed by #690
